enhancements/machine-api/control-plane-machine-set.md

```diff
@@ -620,16 +658,29 @@ were to make an action on the Machine, the Control Plane Machine Set Operator wi
 
 #### Installing a ControlPlaneMachineSet within an existing cluster
 
-When adding a ControlPlaneMachineSet to a existing cluster, the end user will need to define the
-`ControlPlaneMachineSet` resource by copying the existing Control Plane Machine ProviderSpecs.
-Once this is copied, they should remove the failure domain and add the desired failure domains to the `FailureDomains`
-field within the ControlPlaneMachineSet spec.
+For existing clusters that are not born with a ControlPlaneMachineSet already defined,
+we will introduce a second "generator" controller alonside the "main" `ControlPlaneMachineSet`
+controller which will be responsible for constantly checking the cluster state (number of control plane machines present in the cluster,
+number of failure domains defined across all the machinesets in the cluster, most up to date control plane machine provider spec) and
+creating/maintaining an up to date `ControlPlaneMachineSet` in the corresponding namespace, with state set to `Inactive`.
+
+If the cluster has been upgraded to a version that supports this feature, they will be able
+to find an automatically generated, `Inactive` `ControlPlaneMachineSet` with name `cluster` in the `openshift-machine-api` namespace.
+They will need to inspect the generated `ControlPlaneMachineSet` and check that the generated spec matches the desired one,
+optionally modifying the fields to the desired values before then proceeding with its activation, by
+setting the `.spec.state` field to `Active`.
 
-To ensure adding a `ControlPlaneMachineSet` to the cluster is safe, we will need to ensure via a validating webhook
-that the replica count defined in the spec is consistent with the actual size of the control plane within the cluster.
-We will also validate that the failure domains align with those within the cluster already, this will prevent users
+To ensure the values set for the `ControlPlaneMachineSet` are safe for the cluster, we will need to
+validate that the failure domains align with those within the cluster already, this will prevent users
 from accidentally migrating their Control Plane Machines from multiple availability zones to a single zone.
 
+Once the `ControlPlaneMachineSet`'s `.spec.state` will be set to `Active` the main control plane machine set
+controller will start its reconciliation and from then on it will take all the necessary actions to drive
+the cluster control plane machines towards the desired state.
+Once `Active` the `ControlPlaneMachineSet` can't be deactivated anymore. If the user desires to stop the controller
+from reconciling cluster control plane machines, they will need to delete the `ControlPlaneMachineSet`. Doing so
+will stop all the reconciliation operations.
```

Comment:
Might want to add a bit about "If the user wishes to re-activate the CPMS later, the generator will maintain an `Inactive` CPMS ready for them to re-activate"

Reply:
👍  added.